pkg/scheduler/framework/frameworkutils.go

```diff
@@ -70,3 +76,106 @@ func classifyBindings(policy *fleetv1beta1.ClusterSchedulingPolicySnapshot, bind
+//   - bindings that should be deleted, i.e., mark a binding as unschedulable if its target cluster is no
+//     longer picked in the current run.
+//
+// Note that this function will return bindings with all fields fulfilled/refreshed, as applicable.
+func crossReferencePickedCustersAndObsoleteBindings(
+	crpName string,
+	policy *fleetv1beta1.ClusterSchedulingPolicySnapshot,
+	picked ScoredClusters,
+	obsolete []*fleetv1beta1.ClusterResourceBinding,
+) (toCreate, toUpdate, toDelete []*fleetv1beta1.ClusterResourceBinding, err error) {
+	// Pre-allocate with a reasonable capacity.
+	toCreate = make([]*fleetv1beta1.ClusterResourceBinding, 0, len(picked))
+	toUpdate = make([]*fleetv1beta1.ClusterResourceBinding, 0, 20)
+	toDelete = make([]*fleetv1beta1.ClusterResourceBinding, 0, 20)
+
+	// Build a map of picked scored clusters for quick lookup.
+	pickedMap := make(map[string]*ScoredCluster)
+	for _, scored := range picked {
+		pickedMap[scored.Cluster.Name] = scored
+	}
+
+	// Build a map of all clusters that have been cross-referenced.
+	checked := make(map[string]bool)
+
+	for _, binding := range obsolete {
+		scored, ok := pickedMap[binding.Spec.TargetCluster]
+		checked[binding.Spec.TargetCluster] = true
+
+		if ok {
+			// The binding's target cluster is picked again in the current run; yet the binding
+			// is originally created/updated in accordance with an out-of-date scheduling policy.
+
+			// Update the binding so that it is associated with the latest score.
+			affinityScore := int32(scored.Score.AffinityScore)
+			topologySpreadScore := int32(scored.Score.TopologySpreadScore)
+			binding.Spec.ClusterDecision = fleetv1beta1.ClusterDecision{
+				ClusterName: scored.Cluster.Name,
+				Selected:    true,
+				ClusterScore: &fleetv1beta1.ClusterScore{
+					AffinityScore:       &affinityScore,
+					TopologySpreadScore: &topologySpreadScore,
+				},
+				Reason: pickedByPolicyReason,
+			}
+
+			// Update the binding so that it is associated with the lastest scheduling policy.
+			binding.Spec.SchedulingPolicySnapshotName = policy.Name
+
+			// Add the binding to the toUpdate list.
+			toUpdate = append(toUpdate, binding)
+		} else {
+			toDelete = append(toDelete, binding)
+		}
+	}
+
+	for _, scored := range picked {
+		if _, ok := checked[scored.Cluster.Name]; !ok {
+			// The cluster is newly picked in the current run; it does not have an associated binding in presence.
+			name, err := uniquename.NewClusterResourceBindingName(crpName, scored.Cluster.Name)
+			if err != nil {
+				// Cannot get a unique name for the binding; normally this should never happen.
+				return nil, nil, nil, controller.NewUnexpectedBehaviorError(fmt.Errorf("failed to cross reference picked clusters and existing bindings: %w", err))
+			}
+			affinityScore := int32(scored.Score.AffinityScore)
+			topologySpreadScore := int32(scored.Score.TopologySpreadScore)
+
+			toCreate = append(toCreate, &fleetv1beta1.ClusterResourceBinding{
```

Comment:
how about creating a new variable for the newBinding to shorten the line?

toCreate = append(toCreate, newBinding)

Reply:
Done.